stl/inc/random

```diff
@@ -2123,7 +2158,15 @@ public:
 private:
     template <class _Engine>
     result_type _Eval(_Engine& _Eng, const param_type& _Par0) const {
-        return static_cast<_Ty>(_CSTD log(_NRAND(_Eng, _Ty1)) / _Par0._Log_1_p);
+        using _Uty = make_unsigned_t<_Ty>;
+        constexpr auto _Ty_max{(numeric_limits<_Ty>::max)()};
+        const auto _Ty1_max{_Float_upper_bound<_Ty1, _Uty>(static_cast<_Uty>(_Ty_max))};

```

Comment:
Template argument deduction should be able to deduce `_Uty` here. Occurs below. (Our usual convention is "don't help the compiler" - i.e. don't provide explicit template arguments when they can be deduced.)